### PR TITLE
Preview attack delay in equipment description

### PR DESCRIPTION
**Pull request by SentientSupper**
Include attack delay when previewing equipment. This is important for rangers especially those who are wearing cursed or ^drain/contam/fragile armor.

Currently wondering if I should add this for damage rating too but I'd like feedback before proceeding with it.

image

### CODE REVIEW

**Comment by PleasingFungus** [comment]
Cool change! Will take a bit of review.

> Currently wondering if I should add this for damage rating too but I'd like feedback before proceeding with it.

Hm, seems complex. Weakly think it'd be better to hold off on that, at least for now.

File: crawl-ref/source/player-act.cc [hunk 1]
```diff
@@ -379,8 +380,8 @@ random_var player::attack_delay_with(const item_def *projectile, bool rescale,
         attk_delay = div_rand_round(attk_delay, 2);
     }
 
-    return rv::max(div_rand_round(attk_delay * player_speed(), BASELINE_DELAY),
-                   random_var(1));
+    return rv::max(ignore_temporary ? attk_delay : div_rand_round(attk_delay * player_speed(),
```
Comment (Cgettys) [inline crawl-ref/source/player-act.cc:383]: This seems to be getting a bit wordy, might be clearer as a if /else statement?

File: crawl-ref/source/player.cc [hunk 2]
```diff
@@ -6597,9 +6597,10 @@ int player::evasion_scaled(int scale, bool ignore_temporary, const actor* act) c
  * @param ac        The player's AC if this item were equipped.
  * @param ev        The player's EV if this item were equipped.
  * @param sh        The player's SH if this item were equipped.
+ * @param ad        The player's attack delay if this item were equipped.
  */
-void player::ac_ev_sh_with_specific_item(int scale, const item_def& new_item,
-                                         int *ac, int *ev, int *sh)
+void player::status_with_specific_item(int scale, const item_def& new_item,
+                                         int *ac, int *ev, int *sh, int *ad)
```
Comment (Cgettys) [inline crawl-ref/source/player.cc:6603]: Feels like this is getting towards meriting a struct for these  4 parameters?

**Comment by Cgettys** [comment]
Overall, mostly just a few nits, nothing I see as horrible.

**Comment by CrawlOdds** [comment]
An updated version of this was eventually merged in 2f29fc7
